crates/wysiwyg/src/composer_model/mentions.rs

```diff
@@ -77,7 +77,16 @@ where
         let (start, end) = self.safe_selection();
         let range = self.state.dom.find_range(start, end);
 
-        let new_node = DomNode::new_mention(url, text, attributes);
+        // use the display text or the presence of a `data-mention-type => at-room` attribute to decide the mention type
+        let new_node = if text == "@room".into()
+            || attributes.iter().any(|(k, v)| {
+                k == &S::from("data-mention-type") && v == &S::from("at-room")
+            }) {
+            DomNode::new_at_room_mention(attributes)
```

Comment:
As this function is able to create an `@room` mention, we might want to reconsider the API, and either:
- make the `url` argument optional, or
- accept an enum (`AtRoom` | `Matrix(text, url)` - the actual types might come from matrix-org/matrix-rich-text-editor#695).

Alternatively, we could add a new function so that we have:
- `do_insert_mention(text, url) // with mandatory args`
- `do_insert_at_room_mention()`

Reply:
I think I'd be tempted to go a different way and have this function call `DomNode::new_mention` with url, display_text, attributes as we currently have  and then have `MentionNode::new` create either a room, user, or at-room mention as appropriate.

What do you think to that solution? 

If you're happy with me adding a note as the work may be further informed by the WIP mentions utils work (which is what I'm picking up next), I'll do that and address in that subsequent PR.